### PR TITLE
fixing interpolation

### LINKED ISSUE #7717: VirtualBox variable interpolation
`packer version`: v1.4.1
`host operating system`: CentOS 7.6 x64 (3.10.0-957.12.2.el7.x86_64)
`VirtualBox version`: 6.0.8 r130520 (Qt5.6.1)

At a glance it appears that the interpolation of variables within a variable (add in https://github.com/hashicorp/packer/pull/7390), is not honored within the virtualbox-iso builder. I've tested this with the vmware-iso builder and it seems to work. Also I generated a quick test using the null builder to verify.

**vb.json**
```
{
  "builders": [
    {
      "boot_command": [
        "<up><wait><tab><wait> ks=http://{{.HTTPIP}}:{{.HTTPPort}}/centos-7.6-x86_64.kickstart.cfg<enter>"
      ],
      "boot_wait": "5s",
      "guest_additions_mode": "disable",
      "guest_os_type": "RedHat_64",
      "headless": true,
      "http_directory": "http",
      "iso_checksum_type": "none",
      "iso_url": "centos-7.6-x86_64.iso",
      "shutdown_command": "echo 'packer' | sudo -S /sbin/halt -h -p",
      "ssh_password": "password",
      "ssh_timeout": "25m",
      "ssh_username": "root",
      "type": "virtualbox-iso",
      "vm_name": "{{user `name`}}"
    }
  ]
}
```

**templates/base-test.json**
```
{
  "name": "base-{{user `operating_system`}}-{{split uuid \"-\" 4}}"
}
```

**operating-systems/centos-7.6-x86_64-test.json**
```
{
  "operating_system": "centos-7.6-x86_64"
}
```
Running this command results in the screenshot below:
`packer build -var-file=operating-systems/centos-7.6-x86_64-test.json -var-file=templates/base-test.json vb.json`

image

Null builder test case:

**test.json**
```
{
  "builders": [
    {
      "ssh_host": "127.0.0.1",
      "ssh_password": "{{user `ssh_password`}}",
      "ssh_username": "{{user `ssh_username`}}",
      "type": "null"
    }
  ],
  "provisioners": [
    {
      "command": "echo {{user `name`}}",
      "type": "shell-local"
    }
  ],
  "variables": {
    "ssh_password": "{{env `SSH_PASSWORD`}}",
    "ssh_username": "{{env `SSH_USERNAME`}}"
  }
}
```

Running the following command `packer build -var-file=operating-systems/centos-7.6-x86_64-test.json -var-file=templates/base-test.json test.json` results in:

```
null output will be in this color.

==> null: Using ssh communicator to connect: 127.0.0.1
==> null: Waiting for SSH to become available...
==> null: Connected to SSH!
==> null: Running local shell script: /tmp/packer-shell440897968
    null: base-centos-7.6-x86_64-c80724bd94ee
Build 'null' finished.

==> Builds finished. The artifacts of successful builds are:
--> null: Did not export anything. This is the null builder
```

I have ran many other tests including using the `variables` stanza in the vb.json file to try and get this variable to be correctly interpolated.

### LINKED ISSUE #7728: Packer 1.4.x still print sensitive variables on console (Debian stretch)
Hi,
I open a new issue as the previous was closed and seemed to concern MacOS (https://github.com/hashicorp/packer/issues/7448)
Packer version : 1.4.0 & 1.4.1
Host platform : Debian stretch
All logs here : https://gist.github.com/Chr0my/fd250482b82795fc400f4cf8d0038d61

Thanks for your help.

### PR DESCRIPTION
**Pull request by SwampDragons**
The original implementation of user var interpolation wasn't working for variables set via the -var-file or -var command line flags.  This rectifies that oversight.

Closes #7717 
Closes #7728

### CODE REVIEW

File: packer/core.go [hunk 1]
```diff
@@ -385,13 +390,9 @@ func (c *Core) init() error {
...
 
 	for _, v := range c.Template.SensitiveVariables {
-		def, err := interpolate.Render(v.Default, ctx)
-		if err != nil {
-			return fmt.Errorf(
-				"error interpolating default value for '%#v': %s",
-				v, err)
-		}
-		c.secrets = append(c.secrets, def)
+		// log.Printf("k is %#v, v is %#v", k, v)
```
Comment (azr) [inline packer/core.go]:
```suggestion
```

File: packer/core_test.go [hunk 2]
```diff
@@ -576,6 +578,74 @@ func TestCore_InterpolateUserVars(t *testing.T) {
...
+		{
+			// tests that we can interpolate from var files when var isn't set in
+			// originating template
+			"build-basic-interpolated.json",
+			map[string]string{
+				"name":   "gotta-{{user `my_var`}}",
+				"my_var": "interpolate-em-all",
+			},
+			map[string]string{
+				"name":   "gotta-interpolate-em-all",
```
Comment (azr) [inline packer/core_test.go]: 🤣
